en/10_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -172,3 +172,7 @@ PHPDocs are not only useful when looking at the source code, but are also used i
 - Prefer the identical `===` operator over the equality `==` operator for comparisons.
 - If you directly reference a third-party dependency in code, then ensure the dependency is required in the module's `composer.json` file.
 - Avoid hardcoding values when there is a method available to dynamically get a value, for example use [`DataObjectSchema::tableName()`](api:SilverStripe\ORM\DataObjectSchema::tableName()) to get the table name for a `DataObject` model rather than hard coding it.
+- Do not use the `self` keyword, e.g. `self::myMethod()` instead use the short-name of the current class e.g. `MyClass::myMethod()` which is functionally equivalent. This avoids a specific class of bug that is introduced when using late static binding downstream from `self`.
+  - The one exception to this rule is traits as there is no way to know the class the trait is applied to.
```

Comment:
```suggestion
  - The one exception to this rule is traits as there is no way to know the class the trait is applied to. In that case, it's fine to use for getting the class name (e.g. `self::class`) but should only be used for calling static methods if you can't refactor the code to avoid needing to use `self`.
```
In other words, even with traits make an effort to avoid using `self` so we can avoid the late-static-binding bug.

Reply:
I've just checked, and though it looks a little strange I've found that `self::class::myMethod()` works the way you'd expect it to - i.e. it calls the method on the class the code is defined in (in this case the class that directly uses the trait), not the class the object is an instance of.

Maybe this should be the recommended way to call static methods using `self` in traits?

Reply:
Have updated and added mention of self::class::myMethod().  I'll update code-writer and run again post merging this other PRs